tools/functions.sh

```diff
@@ -108,6 +108,8 @@ do_curl_tarball() {
     dldir="$(mktempdir)"
     mkdir "${dldir}/${cmd}"
     do_curl "${dldir}/${cmd}.tar.gz" "${url}"
+    ## need to validate flux here because unpacking the tar changes the hash
+    validate_flux "${path}.txt" "${dldir}/${cmd}.tar.gz"
```

Comment:
Odd to call a function named validate_flux in a generic do_curl_tarball function.  We should either call it to validate checksum or move to the calling function.  i.e., in download_dependency - which might be cleaner as we know there that the configuration provided a checksum txt file.

Reply:
I thought about moving it there but then we would have to change where we do the tarbal unpacking. I did think about moving the unpacked tarball then validating it then unpacking it.